modules/percolator/src/main/java/org/elasticsearch/percolator/ExtractQueryTermsService.java

```diff
@@ -70,29 +74,33 @@ private ExtractQueryTermsService() {
 
     /**
      * Extracts all terms from the specified query and adds it to the specified document.
-     * @param query                 The query to extract terms from
-     * @param document              The document to add the extracted terms to
-     * @param queryTermsFieldField  The field in the document holding the extracted terms
-     * @param unknownQueryField     The field used to mark a document that not all query terms could be extracted.
-     *                              For example the query contained an unsupported query (e.g. WildcardQuery).
-     * @param fieldType The field type for the query metadata field
+     * @param query                     The query to extract terms from
+     * @param document                  The document to add the extracted terms to
+     * @param queryTermsFieldField      The field in the document holding the extracted terms
+     * @param unknownQueryField         The field used to mark a document that not all query terms could be extracted.
+     *                                  For example the query contained an unsupported query (e.g. WildcardQuery).
+     * @param fieldType                 The field type for the query metadata field
+     * @param verifiedCandidateField    The field used to mark if query extraction was accurate and candidate matches don't
+     *                                  need to be verified by {@link org.apache.lucene.index.memory.MemoryIndex} at query time.
      */
     public static void extractQueryTerms(Query query, ParseContext.Document document, String queryTermsFieldField,
```

Comment:
maybe we should use an indexed field rather than a doc values field, this should compress better and would be as efficient at searc time?

in particular I am thinking we could use the same field to indicate that extraction was failed, partial (needs MemoryIndex) or complete (does not need MemoryIndex) using 3 different values?